src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -708,9 +708,15 @@ void instrument_spec_assignst::inclusion_check_assertion(
   comment += " is assignable";
   source_location.set_comment(comment);
 
-  dest.add(goto_programt::make_assertion(
+  goto_programt::instructiont to_add = goto_programt::make_assertion(
     inclusion_check_full(car, allow_null_lhs, include_stack_allocated),
-    source_location));
+    source_location);
+  // Record what target is checked.
+  auto &checked_assigns =
+    static_cast<exprt &>(to_add.condition_nonconst().add(ID_checked_assigns));
+  checked_assigns = car.target();
+
+  dest.add(std::move(to_add));
```

Comment:
I'd suggest to first construct the condition of the assertion as a separate `exprt` (including the `ID_checked_assigns`), and then do `dest.add(goto_programt::make_assertion(modified_condition, source_location)`.